lib/components/Provider/index.tsx

```diff
@@ -133,25 +133,25 @@ export const NotificationAPIProvider: React.FunctionComponent<
   const addNotificationsToState = useCallback((notis: InAppNotification[]) => {
     const now = new Date().toISOString();
     setNotifications((prev) => {
+      // Ensure the notifications are an array
+      notis = Array.isArray(notis) ? notis : [];
+
       notis = notis.filter((n) => {
-        const isExpired =
-          n.expDate && new Date(n.expDate * 1000).toISOString() < now;
-        const isFuture =
-          new Date(n.date).getTime() > new Date(now).getTime() + 1000; // Allow for 1 second margin
-        return !isExpired && !isFuture;
+        if (n.expDate && new Date(n.expDate * 1000).toISOString() > now)
```

Comment:
Seems to be reverting the positive you had made before (the 1 second margin)

Reply:
That's what I get for rushing, this now keeps with my refresh fix and also checking for an Array.